addon/object/index.js

```diff
@@ -179,7 +192,17 @@ export const computed = decoratorWithParams(function(target, key, desc, params)
 
     // Use a standard ember computed since getter/setter arrity is restricted,
     // meaning ember-macro-helpers doesn't provide any benefit
-    return emberComputed(...params, { get, set: setter });
+    if (SUPPORTS_NEW_COMPUTED) {
+      return emberComputed(...params, { get, set: setter });
+    } else {
+      return emberComputed(...params, function(key, value) {
```

Comment:
We need a conditional here I believe. Otherwise it's not possible to clobber (we will always be treated like a setter was specified).

Something like:

```
let callback;
if (setter) {
  callback = function (key, value) {
    if (arguments.length > 1) {
      return setter.call(this, key, value);
    }

    return get.call(this);
  }
} else {
  callback = get;
}

return emberComputed(...params, callback);
```